src/object/schema_class_truncator.hpp

```diff
@@ -0,0 +1,48 @@
+/*
+ *
+ * Copyright 2016 CUBRID Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef _SCHEMA_CLASS_TRUNCATOR_HPP_
+#define _SCHEMA_CLASS_TRUNCATOR_HPP_
+
+#include "class_object.h"
+#include "schema_manager.h"
+
+#include <unordered_set>
+
+namespace cubschema
+{
+  /*
+   * In charge of truncating a class.
+   * During that, it can truncate several classes related to the class with FK or partitioning.
+   */
+  class class_truncator
```

Comment:
In my opinion, it'd be better to keep the prefix. Schema Manager deals with not only Class, but also Domain, Constraint, and so on. 'truncate' seems quite obviously an operation for a class, but considering other operations like `sm_rename_class()` and `sm_delete_class_mop()`, it seems necessary to specify the target for the future.

Reply:
@wrlawodms For now, I can't imagine that we will make domain_truncator or constraint_truncator because the behavior of truncate is related to the purpose of truncator class obviously. There is a similar example I think. We don't need to class_replicator and serial_replicator and so on. We just make replicator class and it deals with all kinds of replication. This conversation related to the class design and it can be different by person.

Reply:
Me neither. I can't imagine `doamin_truncator` and `constraint_truncator`, too. But, I can imagine `class_deleter` as an alternative of `sm_delete_class_mop()`, and I think we can't name it just `deleter`. Considering other operations for the class, I think that we should use the prefix.

Reply:
@wrlawodms I don't want to design(make) a class just related to some kind of existing functions. If you make a 'deleter' as a class, then it can extend(include) to class_deleter and index_deleter, and so on. But, I don't want to push you to follow my thought and If you don't agree with me, then you can use the 'class_' prefix without any hesitation. I just want to share my though(opinion) with you and vice versa.